### PR TITLE
Implementing args for indicating file type (#93)

### LINKED ISSUE #93: The tt.py script should allow arguments for specifying/overriding the input and output file type's extension.
Providing arguments for file type for both the input and output files will allow for more flexibility for processing files.

Note - currently the only output file type is ttml.

### PR DESCRIPTION
**Pull request by BobBrownConsulting**
Closes #93

### CODE REVIEW

File: src/main/python/ttconv/tt.py [hunk 1]
```diff
@@ -31,12 +31,45 @@
...
+
+  @staticmethod
+  def str_to_type(file_type_str: str):
+    return {
+          'ttml': FileTypes.TTML,
+          'scc': FileTypes.SCC
+          }.get(file_type_str.lower(), None)    # None is default
+
+  @staticmethod
+  def type_to_str(file_type) -> str:
```
Comment (palemieux) [inline src/main/python/ttconv/tt.py:55]: Can be replaced by `.value`, e.g.:

```
from enum import Enum

class FileType(Enum):
  TTML = "ttml"
  SCC = "scc"

print(FileType.TTML.value)
```

File: src/main/python/ttconv/tt.py [hunk 2]
```diff
@@ -31,12 +31,45 @@
...
 LOGGER = logging.getLogger(__name__)
 
+class FileTypes(Enum):
+  '''Enumerates the types of supported'''
+  TTML = "ttml"
+  SCC = "scc"
+
+  @staticmethod
+  def str_to_type(file_type_str: str):
+    return {
```
Comment (palemieux) [inline src/main/python/ttconv/tt.py:49]: Can be replaced by:

```
FileType("ttml")
```

Note that it can throw `ValueError`

**Review by palemieux — CHANGES_REQUESTED** [review]
Fix linting issues in tt.py:

```
src\main\python\ttconv\tt.py:158:0: W0311: Bad indentation. Found 8 spaces, expected 6 (bad-indentation)
src\main\python\ttconv\tt.py:177:0: W0311: Bad indentation. Found 8 spaces, expected 6 (bad-indentation)
src\main\python\ttconv\tt.py:48:2: C0116: Missing function or method docstring (missing-function-docstring)
src\main\python\ttconv\tt.py:55:2: C0116: Missing function or method docstring (missing-function-docstring)
src\main\python\ttconv\tt.py:62:2: C0116: Missing function or method docstring (missing-function-docstring)
src\main\python\ttconv\tt.py:63:4: R1705: Unnecessary "else" after "return" (no-else-return)
src\main\python\ttconv\tt.py:64:37: R0123: Comparison to literal (literal-comparison)
src\main\python\ttconv\tt.py:35:0: W0611: Unused import typing (unused-import)
```

File: src/main/python/ttconv/tt.py (hunk 1 above)
Comment (BobBrownConsulting) [reply src/main/python/ttconv/tt.py]: Done

File: src/main/python/ttconv/tt.py (hunk 2 above)
Comment (BobBrownConsulting) [reply src/main/python/ttconv/tt.py]: Done

File: src/main/python/ttconv/tt.py [hunk 3]
```diff
@@ -31,12 +31,45 @@
 from argparse import ArgumentParser
 import xml.etree.ElementTree as et
 from pathlib import Path
+from enum import Enum
+import typing
 import ttconv.imsc.reader as imsc_reader
 import ttconv.imsc.writer as imsc_writer
 import ttconv.scc.reader as scc_reader
 
 LOGGER = logging.getLogger(__name__)
```
[the remark this answers is not in the document]
Comment (BobBrownConsulting) [reply src/main/python/ttconv/tt.py:41]: Discussed with @palemieux the current implementation is sufficient.

**Review by palemieux — APPROVED** [review]

**Pull request merged**